core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2500,6 +2549,367 @@ public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorCo
+                }
+            }
+
+            return setExpressionType(node, returnedType);
+        }
+
+        private List<Type> analyzeJsonPathInvocation(String functionName, Expression node, JsonPathInvocation jsonPathInvocation, StackableAstVisitorContext<Context> context)
+        {
+            // ANALYZE THE CONTEXT ITEM
+            // analyze context item type
+            Expression inputExpression = jsonPathInvocation.getInputExpression();
+            Type inputType = process(inputExpression, context);
+
+            // resolve function to read the context item as JSON
+            JsonFormat inputFormat = jsonPathInvocation.getInputFormat();
+            ResolvedFunction inputFunction = getInputFunction(inputType, inputFormat, inputExpression);
+            Type expectedType = inputFunction.getSignature().getArgumentType(0);
+            coerceType(inputExpression, inputType, expectedType, format("%s function input argument", functionName));
+            jsonInputFunctions.put(NodeRef.of(inputExpression), inputFunction);
+
+            // ANALYZE JSON PATH PARAMETERS
+            // TODO verify parameter count? Is there a limit on Row size?
+
+            ImmutableMap.Builder<String, Type> types = ImmutableMap.builder(); // record parameter types for JSON path analysis
+            Set<String> uniqueNames = new HashSet<>(); // validate parameter names
+
+            // this node will be translated into a FunctionCall, and all the information it carries will be passed as arguments to the FunctionCall.
+            // all JSON path parameters are wrapped in a Row, and constitute a single FunctionCall argument.
+            ImmutableList.Builder<RowType.Field> fields = ImmutableList.builder();
+
+            List<JsonPathParameter> pathParameters = jsonPathInvocation.getPathParameters();
+            for (JsonPathParameter pathParameter : pathParameters) {
+                Expression parameter = pathParameter.getParameter();
+                String parameterName = pathParameter.getName().getCanonicalValue();
+                Optional<JsonFormat> parameterFormat = pathParameter.getFormat();
+
+                // type of the parameter passed to the JSON path:
+                // - parameters of types numeric, string, boolean, date,... are passed as-is
+                // - parameters with explicit or implicit FORMAT, are converted to JSON (type JSON_2016)
+                // - all other parameters are cast to VARCHAR
+                Type passedType;
+
+                if (!uniqueNames.add(parameterName)) {
+                    throw semanticException(DUPLICATE_PARAMETER_NAME, pathParameter.getName(), "%s JSON path parameter is specified more than once", parameterName);
+                }
+
+                if (parameter instanceof LambdaExpression || parameter instanceof BindExpression) { // TODO do we need this limitation?
```

Comment:
FunctionType is not an SQL data type. I wanted to limit the accepted parameters to data types. I don't want to rely on the coercion rules, since those might change. I think I should just remove the TODO. What do you think?

Reply:
YEah - please remove TODO.